plugins/arcgis/web-app/projects/main/src/lib/arc-layer/arc-layer.component.ts

```diff
@@ -71,8 +78,21 @@ export class ArcLayerComponent implements OnInit {
     return this.layers.length == 0;
   }
 
-  inputChanged(layerUrl: string, token?: string) {
-    const url = this.addToken(layerUrl, token);
+  inputChanged(layerUrl: string, token?: string, username?: string, password?: string) {
+    let url: string;
+     //TODO - switch to username/pw being in body and avoid in url query
+     //TODO - remove hardcoded username/pw and update UI to provide
```

Comment:
@newmanw Please search through this code for //TODO. Comment on any you don't feel worth doing.